docs/scripts/getTypeScriptProjects.ts

```diff
@@ -0,0 +1,106 @@
+import path from 'path';
+import * as ts from 'typescript';
+import { Project, ProjectNames, Projects } from './api/utils';
+
+const workspaceRoot = path.resolve(__dirname, '../../');
+
+interface CreateProgramOptions {
+  name: ProjectNames;
+  rootPath: string;
+
+  /**
+   * Config to us to build this package
+   * The path must be relative to the root path.
+   */
+  tsConfigPath: string;
+
+  /**
+   * File used as root of the package
+   * The path must be relative to the root path.
+   */
+  entryPointPath: string;
+
+  /**
+   * Folder containing all the components of this package
+   * The path must be relative to the root path.
+   */
+  componentsFolder?: string;
+
+  /**
+   * Additional files containing components outside the component's folder
+   * The path must be relative to the root path.
+   */
+  otherComponentFiles?: string[];
+}
+
+const createProject = (options: CreateProgramOptions): Project => {
+  const { name, tsConfigPath, rootPath, entryPointPath, componentsFolder, otherComponentFiles } =
+    options;
+
+  const compilerOptions = ts.parseJsonConfigFileContent(
+    ts.readConfigFile(path.join(rootPath, tsConfigPath), ts.sys.readFile).config,
+    ts.sys,
+    rootPath,
+  );
+
+  const fullEntryPointPath = path.join(rootPath, entryPointPath);
+
+  const program = ts.createProgram({
+    rootNames: [fullEntryPointPath],
+    options: compilerOptions.options,
+  });
+
+  const checker = program.getTypeChecker();
+  const sourceFile = program.getSourceFile(fullEntryPointPath);
+
+  const exports = Object.fromEntries(
+    checker.getExportsOfModule(checker.getSymbolAtLocation(sourceFile!)!).map((symbol) => {
+      return [symbol.name, symbol];
+    }),
+  );
+
+  return {
+    name,
+    exports,
+    program,
+    checker,
+    workspaceRoot,
+    componentsFolder:
+      componentsFolder === undefined ? undefined : path.join(rootPath, componentsFolder),
+    otherComponentFiles:
+      otherComponentFiles === undefined
+        ? undefined
+        : otherComponentFiles.map((file) => path.join(rootPath, file)),
```

Comment:
To avoid line breaks.

```suggestion
    componentsFolder: componentsFolder ? path.join(rootPath, componentsFolder) : undefined,
    otherComponentFiles: otherComponentFiles?.map((file) => path.join(rootPath, file)),
```